hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/blockmanagement/TestReplicationPolicyExcludeSlowNodes.java

```diff
@@ -176,4 +185,58 @@ public void testSlowPeerTrackerEnabledClearSlowNodes() throws Exception {
     }
   }
 
+  /**
+   * Dependent on the SlowNode related config, therefore placing
+   * 'testCollectSlowNodesIpAddrFrequencyMetrics' unit test in the
+   * TestReplicationPolicyExcludeSlowNodes class.
+   * <br>
+   * Test metrics associated with CollectSlowNodesIpAddrFrequency.
+   */
+  @Test
+  public void testCollectSlowNodesIpAddrFrequencyMetrics() throws Exception {
+    namenode.getNamesystem().writeLock();
+    try {
+      FSNamesystem fsNamesystem = namenode.getNamesystem();
+      assertEquals("{}", fsNamesystem.getCollectSlowNodesIpAddrFrequencyMap(), "{}");
+      MBeanServer mBeanServer = ManagementFactory.getPlatformMBeanServer();
+      ObjectName mxBeanName = new ObjectName("Hadoop:service=NameNode,name=FSNamesystemState");
+      String ipAddrFrequency =
+          (String) mBeanServer.getAttribute(mxBeanName, "CollectSlowNodesIpAddrFrequencyMap");
+      assertEquals("{}", ipAddrFrequency, "{}");
+
+      // add nodes
+      for (DatanodeDescriptor dataNode : dataNodes) {
+        dnManager.addDatanode(dataNode);
+      }
+
+      // mock slow nodes
+      SlowPeerTracker tracker = dnManager.getSlowPeerTracker();
+      Assert.assertNotNull(tracker);
+      OutlierMetrics outlierMetrics = new OutlierMetrics(0.0, 0.0, 0.0, 5.0);
+      tracker.addReport(dataNodes[0].getInfoAddr(), dataNodes[2].getInfoAddr(), outlierMetrics);
+      tracker.addReport(dataNodes[1].getInfoAddr(), dataNodes[2].getInfoAddr(), outlierMetrics);
+
+      // waiting for slow nodes collector run and collect at least 2 times
+      Thread.sleep(3000);
```

Comment:
If we want to wait for a while, we use `GenericTestUtils.waitFor(..)` instead of `thread.sleep`.

Reply:
> If we want to wait for a while, we use `GenericTestUtils.waitFor(..)` instead of `thread.sleep`.

Thanks @slfan1989, I have fixed it.